### PR TITLE
Update megacity records

### PR DESCRIPTION
**Pull request by stepps00**
Fixes a portion of https://github.com/whosonfirst-data/whosonfirst-data/issues/1547

- Updates megacity records with new:
  - concordances
  - names (if not already present)
  - label centroids
  - megacity flags
- Updates geometries of any megacity record with a `Point` geometry
- Updates parent geometries when necessary

### CODE REVIEW

**Comment by stepps00** [comment]
@nvkelso here's an example PR with both property and geometry updates to megacity records (with newer name logic, not to overwrite existing names).

File: data/101/912/923/101912923.geojson [hunk 1]
```diff
@@ -517,10 +518,10 @@
...
...862],[30.728617,36.940376],[30.729703,36.940132],[30.728904,36.937518],[30.727818,36.937761],[30.727552,36.93689],[30.726466,36.937133],[30.726199,36.936262],[30.725113,36.936505],[30.724049,36.933019],[30.722963,36.933263],[30.722165,36.930648],[30.723251,36.930405],[30.722985,36.929533],[30.725157,36.929046],[30.724891,36.928175],[30.732493,36.92647],[30.73276,36.927341],[30.733846,36.927098],[30.734644,36.929712],[30.731386,36.930443],[30.732184,36.933058],[30.733271,36.932814],[30.734602,36.937171],[30.733515,36.937415],[30.734314,36.94003],[30.739745,36.938811],[30.740012,36.939683],[30.74327,36.938952],[30.744602,36.943309],[30.743516,36.943553],[30.743782,36.944424],[30.744869,36.94418],[30.745135,36.945052],[30.746222,36.944808],[30.745955,36.943937],[30.747041,36.943693],[30.746775,36.942822],[30.75112,36.941846],[30.751919,36.944461],[30.750833,36.944704],[30.7511,36.945576],[30.750013,36.94582],[30.75028,36.946691],[30.740503,36.948885],[30.740236,36.948013],[30.73915,36.948257],[30.738884,36.947386],[30.737797,36.947629],[30.736998,36.945015],[30.732653,36.945989],[30.733185,36.947732],[30.734272,36.947489],[30.735337,36.950975],[30.725559,36.953167]]],"type":"Polygon"}
```
Comment (nvkelso) [inline data/101/912/923/101912923.geojson:526]: Should this geometry get an update? It looks to be the old Quaatroshapes geom still, not including the airport or the left side of the city (I don't know what it should really look like).

**Comment by nvkelso** [comment]
Generally looks good!

But let's also add polygon for `Istanbul`.

**Review by nvkelso — CHANGES_REQUESTED** [review]
See comments

File: data/101/912/923/101912923.geojson (hunk 1 above)
Comment (stepps00) [reply data/101/912/923/101912923.geojson]: I've updated this locality with a more accurate polygon (which includes the airport) in 3c51e65.

**Comment by stepps00** [comment]
> But let's also add polygon for Istanbul.

**Pull request merged**